jupyterhub/utils.py

```diff
@@ -85,6 +85,7 @@ def make_ssl_context(keyfile, certfile, cafile=None, verify=True, check_hostname
         return None
     purpose = ssl.Purpose.SERVER_AUTH if verify else ssl.Purpose.CLIENT_AUTH
     ssl_context = ssl.create_default_context(purpose, cafile=cafile)
+    ssl_context.load_default_certs()
```

Comment:
Is this dependent on the `purpose`? https://docs.python.org/3/library/ssl.html#ssl.SSLContext.load_default_certs says it can be `Purpose.SERVER_AUTH` (default) and `Purpose.CLIENT_AUTH`?

Reply:
It is dependent, though in my case I only need it to load server certs.

That being said, `verify` isn't ever specified that I can tell when calling `make_ssl_context` in the the code base. so it's always `SERVER_AUTH` I think. Additionally, when I reviewed the history of changes related to internal_ssl, it seems that it was added without ever being used.

I'd say we could probably remove `purpose` and `verify` entirely from the function. What do you think?

Reply:
The simpler fix is to pass the purpose arg to this function:

```suggestion
        ssl_context.load_default_certs(purpose)
```

that way we can still use this function for creating client_auth contexts.

Reply:
Do we even need the `verify`/`purpose` anymore? It doesn't like `verify` is ever passed as an argument, so it looks like it's just dead code at this point.

Reply:
@chancez Apologies for dropping this. I'll let @minrk confirm as he knows this better, but I think this is considered a public function that could be used by external components, so removing a parameter is an API change.

Reply:
I updated it to take purpose. I think removal could be a separate change from this PR anyways.